src/components/AddEventToPublishedScheduleForm/AddEventToPublishedScheduleForm.jsx

```diff
@@ -1,133 +1,358 @@
-  import * as yup from 'yup';
-  
-  const schema = yup.object({
-      confirmed: yup.boolean().default(true).required("Confirmation required"),
-      startTime: yup.date().required('Start time required'),
-      endTime: yup.date().required('End time required').min(yup.ref('startTime'), 'End time must be after start time'),
-      cohort: yup.number().required('Cohort required').min(2000),
-      notes: yup.string().nullable()
+  Box,
+  FormLabel,
+  Input,
+  FormControl,
+  FormErrorMessage,
+  Button,
+  Textarea,
+  Checkbox,
+  useToast,
+  Heading,
+  Flex,
+  Text,
+  Stack
+} from '@chakra-ui/react';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+import { useContext, useEffect, useState } from 'react';
+import { NPOBackend } from '../../utils/auth_utils';
+import { DayIdContext } from '../../pages/PublishedSchedule/AddDayContext';
+import * as yup from 'yup';
+import {
+  seasonOptions,
+  yearOptions,
+  subjectOptions,
+  eventOptions,
+} from '../Catalog/SearchFilter/filterOptions';
+import useSearchFilters from '../Catalog/SearchFilter/useSearchFilters';
+import Dropdown from '../Dropdown/Dropdown';
+import PropTypes from 'prop-types';
+import { PlannerContext } from '../Planner/PlannerContext';
+import PlannedEvent, { convertTimeToMinutes } from '../Planner/PlannedEvent';
+
+const schema = yup.object({
+    startTime: yup.string().required('Start time is required'),
+    endTime: yup.string()
+      .required('End time is required')
+      .test('is-after', 'End time must be after start time', function(endTime) {
+        const startTime = this.parent.startTime;
+        return startTime && endTime && startTime < endTime;
+      }),
+    host: yup.string().max(50, 'Host exceeds 50 character limit').default('').nullable(),
+    title: yup.string().required('Title Required').max(50, 'Title exceeds 50 character limit'),
+    description: yup
+      .string()
+      .max(256, 'Description exceeds 256 character limit')
+      .default('')
+      .nullable(),
+    tentative: yup.boolean()
+});
+
+const AddEventToPublishedScheduleForm = ({ cancelFunction, eventData }) => {
+  const { plannedEventsContext } = useContext(PlannerContext);
+  const [plannedEvents, setPlannedEvents] = plannedEventsContext;
+  const { filters, filterValues } = useSearchFilters();
+  const [seasonFilter, yearFilter, subjectFilter, eventFilter] = filters;
+  const { dayId } = useContext(DayIdContext);
+
+  const [formData, setFormData] = useState({tentative: false});
+
+  useEffect(() => {
+    if (formData.startTime && formData.endTime) {
+      const newPlannedEvent = new PlannedEvent(
+        -1,
+        formData.title,
+        convertTimeToMinutes(formData.startTime),
+        convertTimeToMinutes(formData.endTime),
+        formData.host,
+        formData.tentative ? true : false
+      )
+      setPlannedEvents([...plannedEvents.filter(e => e.id != -1), newPlannedEvent])
+    }
+  }, [formData])
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'plannedEvents' and 'setPlannedEvents'. Either include them or remove the dependency array.